packages/talos_messenger_core/src/talos_messenger_service.rs

```diff
@@ -0,0 +1,46 @@
+use futures_util::future::try_join_all;
+
+use crate::{
+    core::MessengerSystemService,
+    errors::{MessengerServiceError, MessengerServiceErrorKind, MessengerServiceResult},
+};
+
+pub struct TalosMessengerService {
+    pub services: Vec<Box<dyn MessengerSystemService + Send + Sync>>,
+}
+
+impl TalosMessengerService {
+    pub async fn run(self) -> MessengerServiceResult {
+        let service_handle = self.services.into_iter().map(|mut service| tokio::spawn(async move { service.run().await }));
+
+        let k = try_join_all(service_handle).await.map_err(|e| MessengerServiceError {
```

Comment:
Do we need this join all? By reading I understand there is only one `service_handle`.

Reply:
The tasks are mapped from an iterator, so probably the variable should be `service_handles` (plural).

Reply:
yes, should be `service_handles`